lib/SIL/IR/SILPrinter.cpp

```diff
@@ -816,6 +816,12 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
       for (auto Id : PredIDs)
         *this << ' ' << Id;
     }
+
+    // If the basic block has a name available, print it as well
+    auto debugName = BB->getDebugName();
+    if (debugName.hasValue()) {
+      *this << " /// " << debugName.getValue();
+    }
```

Comment:
open to other ideas where to print it; wasn't quite sure how it is printed in IR, I suppose we could make it the same as there?

Reply:
I would print this the same way we do for demangled names--print a separate comment line *before* the block ID.

As long as the .sil still parses, it's a matter of taste. You can check by piping the output back through swift-frontend or sil-opt.

Reply:
That sounds good, will do 👍